senders/opsgenie/init.go

```diff
@@ -0,0 +1,58 @@
+package opsgenie
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/moira-alert/moira"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/alert"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/client"
+)
+
+// Sender implements the Sender interface for opsgenie
+type Sender struct {
+	apiKey               string
+	client               *alert.Client
+	logger               moira.Logger
+	location             *time.Location
+	ImageStores          map[string]moira.ImageStore
+	imageStoreID         string
+	imageStore           moira.ImageStore
+	imageStoreConfigured bool
+	frontURI             string
+}
+
+// Init initializes the opsgenie sender
+func (sender *Sender) Init(senderSettings map[string]string, logger moira.Logger, location *time.Location, dateTimeFormat string) error {
+	var ok bool
+
+	if sender.apiKey, ok = senderSettings["api_key"]; !ok {
+		return fmt.Errorf("cannot read the api_key from the sender settings")
+	}
+
+	sender.imageStoreID, ok = senderSettings["image_store"]
```

Comment:
Yes. All senders using image store will have to have this snippet. I moved it to the senders package so that both opsgenie and pagerduty can use it. Also added tests!